### ISSUE TITLE
Human-friendly errors status update?

### ISSUE THREAD

**Issue by Latros** [balderdashy/sails#2140]
Hey folks - one common-thread-of-annoyance that I've had from many-a-sails-app over the last year has been validation errors and their respective messages. Specifically: the ability to override the "programmer-friendly" validation errors like `undefined should be a string (instead of \"null\", which is a object)` with our own custom errors like `[Input] is a required field`.

It was originally on the forecast for "0.10.0-rc6", which has now (as far as the Trello indicates) been pushed back to "0.10.x".

I am interested in helping get this out the door, or at the very least being able to fork sails and edit my own copy if you've already got it under way :)

So, how goes this functionality?

**Also**, I'm interested in attempting this on my own regardless, would one of you Sails devs (or @robwormald) know where I could find the place in the code where the actual response containing the `WLValidationError` is sent? I poked around but couldn't find it :)

Edit: forgot that I made a repo and might be making a library (separate from sails) thats a human-friendly-error-producing validation library. [Take a look at my proposed implementation](https://github.com/Latros/friendly-validator.js), that's the format that I think the responses should be sent back to the client.

**Comment by lykmapipo** [comment]
For sails v0.11.X you may use this [sails-hook-validation](https://github.com/lykmapipo/sails-hook-validation)

**Comment by derekbasch** [comment]
Neat stuff @lykmapipo .

**Comment by lykmapipo** [comment]
@derekbasch +1

**Comment by byrnedo** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @Latros.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
